private-templates-service/client/src/components/Common/VersionModal/VersionModal.tsx

```diff
@@ -0,0 +1,155 @@
+      dispatch(updateTemplate(undefined, version, templateJSON, undefined, undefined, state, undefined));
+    },
+    deleteTemplateVersion: (version: string) => {
+      dispatch(deleteTemplateVersion(version, undefined))
+    }
+  }
+}
+
+interface State {
+  versionList: boolean[];
+}
+
+class VersionModal extends React.Component<Props, State> {
+  constructor(props: Props){
+    super(props);
+    this.state = { versionList: this.props.template.instances? new Array(this.props.template.instances.length) : new Array() }
+  }
+  
+  delete = () => {
+    let list = this.props.template.instances!;
+    for (let i = 0; i < list.length; i++) {
+      if (!this.state.versionList[i]) continue;
+      this.props.deleteTemplateVersion(list[i].version!);
+      this.props.closeModal();
+    }
+  }
+
+  publish = () => {
+    let list = this.props.template.instances!;
+    for (let i = 0; i < list.length; i++) {
+      if (!this.state.versionList[i]) continue;
+      this.props.updateTemplateState(PostedTemplate.StateEnum.Live, list[i].version!, list[i].json );
+      this.props.closeModal();
+    }
+  }
+
+  unpublish = () => {
+    let list = this.props.template.instances!;
+    for (let i = 0; i < list.length; i++) {
+      if (!this.state.versionList[i]) continue;
+      this.props.updateTemplateState(PostedTemplate.StateEnum.Deprecated, list[i].version!, list[i].json);
+      this.props.closeModal();
+    }
+  }
+
+  render() {
+    const { template } = this.props;
+    return (
+      <BackDrop>
+        <Modal>
+          <Header>Versions</Header>
+          <Description>There are {template.instances!.length} versions for <DescriptionAccent>{template.name}</DescriptionAccent></Description>
+          <CenterPanelWrapper>
+            <Card>
+              <CardHeaderRow>
+                <CardHeaderText>Version</CardHeaderText>
+                <CardHeaderText>Published</CardHeaderText>
+                <CardHeaderText>Status</CardHeaderText>
+                <CardHeaderText>{`${this.state.versionList.filter(function(s) { return s; }).length} Selected`}</CardHeaderText>
+              </CardHeaderRow>
+              <CardBody>
+              {this.props.template.instances && this.props.template.instances.map((instance: TemplateInstance, index: number) => (
+            <VersionCardRow>
+              <VersionWrapper>
+                {instance.version}
+              </VersionWrapper>      
+              <DateWrapper>{instance.publishedAt? getDateString(instance.publishedAt) : "Not published"}</DateWrapper>
+              <StatusWrapper>
+                <StatusIndicator state={instance.state}/>
+                <Status>{instance.state && instance.state.toString().charAt(0).toUpperCase() + instance.state.toString().slice(1)}</Status>
+              </StatusWrapper>
+              <CheckboxWrapper><Checkbox defaultChecked={false} onChange={(e, checked) => {
```

Comment:
Can you use the `checked` prop for the checkbox component to set its value ? defaultChecked can be removed and OnChange can then be 
`
let updatedVersion = this.state.versionList;
updatedVersion[index] = !updatedVersion[index]
this.setState({versionList: updatedVersion});
`

Reply:
see the `checked` prop in  https://developer.microsoft.com/en-us/fabric#/controls/web/checkbox

Reply:
![image](https://user-images.githubusercontent.com/19780117/76452843-009d3f80-638f-11ea-9d01-c0fa8592e6e4.png)